src/org/opencms/main/CmsShell.java

```diff
@@ -674,6 +675,13 @@ public void execute(Reader reader) {
                     m_out.println(line);
                     continue;
                 }
+                // In linux, the up and down arrows generate escape sequences that cannot be properly handled.
+                // If a escape sequence is detected, OpenCms prints a warning message
+                if (line.indexOf(KeyEvent.VK_ESCAPE) != -1) {
```

Comment:
I only have a linux to test so I cannot asses if this additional protection make sense under all platforms.